### PR TITLE
adding altID parameter to refresh endpoint and removing productName

### PR DESCRIPTION
**Pull request by dcanar9**
made changes to the refresh process, the white source collector items are now found using either productToken or altIdentifier

### CODE REVIEW

**Comment by dcanar9** [comment]
Compile error because Im going to use a new version of core to get collectorItems from the repository

File: src/main/java/com/capitalone/dashboard/collector/DefaultWhiteSourceClient.java [hunk 1]
```diff
@@ -673,10 +675,19 @@ private void associateBuildToLibraryPolicy(String buildUrl, String clientReferen
...
         Collector collector = collectorRepository.findByName(Constants.WHITE_SOURCE);
         Map<String, Object> options = getOptions(orgName, projectToken);
-        Iterable<CollectorItem> collectorItems = collectorItemRepository.findAllByOptionMapAndCollectorIdsIn(options, Stream.of(collector.getId()).collect(Collectors.toList()));
+        Iterable<CollectorItem> collectorItems = new ArrayList<>();
+
+        if (StringUtils.isNotEmpty(projectToken)) {
+            collectorItems = collectorItemRepository.findAllByOptionMapAndCollectorIdsIn(options, Stream.of(collector.getId()).collect(Collectors.toList()));
+        }
+        else if (StringUtils.isNotEmpty(altIdentifier)) {
+            collectorItems = collectorItemRepository.findAllByOptionMapCollectorIdAndAltId(options, collector.getId(), altIdentifier);
```
Comment (nameisaravind) [inline src/main/java/com/capitalone/dashboard/collector/DefaultWhiteSourceClient.java:687]: lookup by collectorId and altIdentifier alone. @dcanar9

**Review by nameisaravind — COMMENTED** [review]
this refresh flow should only depends on either projectToken or altIdentifier. orgName shouldn't be used. plz update as discussed

File: src/main/java/com/capitalone/dashboard/collector/DefaultWhiteSourceClient.java [hunk 2]
```diff
@@ -413,27 +413,23 @@ public Map<String, WhiteSourceProjectVital> getOrgProjectVitals(WhitesourceOrg w
...
-        components.forEach(component -> {
-            LibraryPolicyResult libraryPolicyResult = getProjectAlerts(component, null, whiteSourceSettings.getWhiteSourceServerSettings().get(0));
-            if (Objects.nonNull(libraryPolicyResult)) {
-                libraryPolicyResult.setCollectorItemId(component.getId());
-                LibraryPolicyResult libraryPolicyResultExisting = getLibraryPolicyData(component, libraryPolicyResult);
-                if (Objects.nonNull(libraryPolicyResultExisting)) {
-                    libraryPolicyResult.setId(libraryPolicyResultExisting.getId());
+    public void refresh (String projectToken, String altIdentifier){
+        List<WhiteSourceComponent> components = getWhiteSourceComponents(projectToken, altIdentifier);
+        if(!components.isEmpty()) {
```
Comment (nameisaravind) [inline src/main/java/com/capitalone/dashboard/collector/DefaultWhiteSourceClient.java:420]: this is redundant, not necessary

File: src/main/java/com/capitalone/dashboard/model/WhiteSourceRefreshRequest.java [hunk 3]
```diff
@@ -4,26 +4,15 @@
 
 public class WhiteSourceRefreshRequest {
 
-    @ApiModelProperty(notes = "WhiteSource Organization name",name="OrgName",required=true)
```
Comment (nameisaravind) [inline src/main/java/com/capitalone/dashboard/model/WhiteSourceRefreshRequest.java:7]: let's not delete orgName and productName for now, lets keep them required false.

**Review by nameisaravind — CHANGES_REQUESTED** [review]

File: src/main/java/com/capitalone/dashboard/model/WhiteSourceRefreshRequest.java (hunk 3 above)
Comment (dcanar9) [reply src/main/java/com/capitalone/dashboard/model/WhiteSourceRefreshRequest.java:7]: 👍 placed them back in the request obj

**Review by nameisaravind — DISMISSED** [review]
lgtm

**Review by aswink19 — APPROVED** [review]
LGTM

**Review by aswink19 — APPROVED** [review]
LGTM

**Pull request merged**